src/protagonist/DLCS.AWS/Cloudfront/CacheInvalidator.cs

```diff
@@ -0,0 +1,57 @@
+﻿using Amazon.CloudFront;
+using Amazon.CloudFront.Model;
+using DLCS.AWS.Settings;
+using DLCS.Core;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+
+namespace DLCS.AWS.Cloudfront;
+
+public class CacheInvalidator : ICacheInvalidator
+{
+    private readonly ILogger<CacheInvalidator> logger;
+    private readonly IAmazonCloudFront client;
+    private CloudfrontSettings cloudfrontSettings;
+    
+    
+    public CacheInvalidator(
+        ILogger<CacheInvalidator> logger,
+        IAmazonCloudFront client,
+        IOptions<AWSSettings> settings)
+    {
+        this.logger = logger;
+        this.client = client;
+        cloudfrontSettings = settings.Value.Cloudfront;
+    }
+    
+    public async Task<bool> InvalidateCdnCache(List<string> invalidationPaths, CancellationToken cancellationToken)
+    {
+        var invalidationRequest = new CreateInvalidationRequest
+        {
+            DistributionId = cloudfrontSettings.DistributionId,
+            InvalidationBatch = new InvalidationBatch
+            {
+                Paths = new Paths
+                {
+                    Quantity = invalidationPaths.Count,
+                    Items = invalidationPaths
+                },
+                CallerReference = DateTime.Now.Ticks.ToString()
```

Comment:
I wonder how long this lasts and whether it's worth putting `assetId` in it somewhere to avoid duplicate requests that are close together. Maybe assetId and sqs messageId or something. (probably not worth doing now, can revisit if we're seeing a lot of duplicates).